src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHost.cs

```diff
@@ -151,18 +153,45 @@ protected override Task InitializeCoreAsync(CancellationToken cancellationToken)
                 await _commonServices.ThreadingService.SwitchToUIThread();
                 var projectManager = GetProjectManager();
 
-                var razorProperties = await _commonServices.ActiveConfiguredProjectRazorProperties.GetRazorGeneralPropertiesAsync().ConfigureAwait(false);
-                var languageVersion = await razorProperties.RazorLangVersion.GetEvaluatedValueAtEndAsync().ConfigureAwait(false);
+                var languageVersion = update.Value.CurrentState[Rules.RazorGeneral.SchemaName].Properties[Rules.RazorGeneral.RazorLangVersionProperty];
+                var defaultConfiguration = update.Value.CurrentState[Rules.RazorGeneral.SchemaName].Properties[Rules.RazorGeneral.RazorDefaultConfigurationProperty];
 
-                if (string.IsNullOrEmpty(languageVersion))
+                RazorConfiguration configuration = null;
+                if (!string.IsNullOrEmpty(languageVersion) && !string.IsNullOrEmpty(defaultConfiguration))
+                {
+                    if (!RazorLanguageVersion.TryParse(languageVersion, out var parsedVersion))
+                    {
+                        parsedVersion = RazorLanguageVersion.Latest;
+                    }
+
+                    var extensions = update.Value.CurrentState[Rules.RazorExtension.PrimaryDataSourceItemType].Items.Select(e =>
+                    {
+                        return new ProjectSystemRazorExtension(e.Key);
+                    }).ToArray();
+
+                    var configurations = update.Value.CurrentState[Rules.RazorConfiguration.PrimaryDataSourceItemType].Items.Select(c =>
+                    {
+                        var includedExtensions = c.Value[Rules.RazorConfiguration.ExtensionsProperty]
+                            .Split(';')
+                            .Select(name => extensions.Where(e => e.ExtensionName == name).FirstOrDefault())
+                            .Where(e => e != null)
+                            .ToArray();
+
+                        return new ProjectSystemRazorConfiguration(parsedVersion, c.Key, includedExtensions);
+                    }).ToArray();
+
+                    configuration = configurations.Where(c => c.ConfigurationName == defaultConfiguration).FirstOrDefault();
```

Comment:
This code turns this https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.props

into a RazorConfiguration